### ISSUE TITLE
SSO External Browser Auth calls back to wrong port sometimes

### ISSUE THREAD

**Issue by archiewood** [evidence-dev/evidence#1290]
## Steps To Reproduce

- Evidence running on `localhost:3000`
- Snowflake reachable and authenticated via external browser at remote URL `REDACTED`
- Callback to evidence was on localhost:33889?token=blah. Note that the port changed upon a restart of Evidence

### Environment

- IDP: Okta
- Dev Environment: `VSCode devcontainer`
- Database: Snowflake
- Authenticator: SSO External Browser
- Evidence Version: 
- Node version (`node -v`):
- npm version (`npm -v`):
- OS:
- Browser:

## Expected Behavior

- Callback to evidence was on `localhost:3000?token=<token>`

## Actual Behaviour

- Callback to evidence was on `localhost:33889?token=<token>`. Note that the port changed upon a restart of Evidence

## Workarounds

- if you change the URL from `localhost:33889?token=<token>` to `localhost:3000?token=<token>` in the browser, then it works, but this is annoying

**Comment by archiewood** [comment]
@jdimmerman

**Comment by bfellers** [comment]
@archiewood I was able to resolve this by changing the following in my .vscode/settings.json (or wherever you prefer to update your vscode settings)

```
    "remote.autoForwardPortsSource": "process", // This is defaulted to "hybrid", which is the main issue
    "remote.autoForwardPorts": true,
    "remote.autoForwardPortsFallback": 0, // This is necessary because if you auto forward more than 20 ports the autoForwardPortsSource switches back to "hybrid"
```

I think it would be nice if the port create/forward/redirect could be removed to just use 3000. But this does work and I think makes using port 3000 as the default return url less of a priority.

**Comment by archiewood** [comment]
Just to be clear - if we added this to the some default settings in our template project, then this would resolve the issue for new users?

https://github.com/evidence-dev/template/tree/main/.vscode

**Comment by bfellers** [comment]
Yes, the workspace settings defined in .vscode/settings.json should take precedence. If those are getting copied into the project they should correctly configure vscode to forward ports happily.

VS Code Docs on this: https://code.visualstudio.com/docs/getstarted/settings#_settings-precedence
